integration/framework.go

```diff
@@ -0,0 +1,301 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	giantclientset "github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/e2e-harness/pkg/harness"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	certOperatorValuesFile = "/tmp/cert-operator-values.yaml"
+	// certOperatorChartValues values required by cert-operator-chart, the environment
+	// variables will be expanded before writing the contents to a file.
+	certOperatorChartValues = `commonDomain: ${COMMON_DOMAIN}
+clusterName: ${CLUSTER_NAME}
+Installation:
+  V1:
+    Auth:
+      Vault:
+        Address: http://vault.default.svc.cluster.local:8200
+        CA:
+          TTL: 1440h
+    Guest:
+      Kubernetes:
+        API:
+          EndpointBase: ${COMMON_DOMAIN}
+    Secret:
+      CertOperator:
+        SecretYaml: |
+          service:
+            vault:
+              config:
+                token: ${VAULT_TOKEN}
+      Registry:
+        PullSecret:
+          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"$REGISTRY_PULL_SECRET\"}}}"
+`
+)
+
+type framework struct {
+	cs   kubernetes.Interface
+	gsCs *giantclientset.Clientset
+}
+
+func newFramework() (*framework, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", harness.DefaultKubeConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	cs, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	gsCs, err := giantclientset.NewForConfig(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	f := &framework{
+		cs:   cs,
+		gsCs: gsCs,
+	}
+	return f, nil
+}
+
+func (f *framework) runningPodFunc(namespace, labelSelector string) func() error {
+	return func() error {
+		pods, err := f.cs.CoreV1().
+			Pods(namespace).
+			List(metav1.ListOptions{
+				LabelSelector: labelSelector,
+			})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(pods.Items) > 1 {
+			return microerror.Mask(tooManyResultsError)
+		}
+		pod := pods.Items[0]
+		phase := pod.Status.Phase
+		if phase != v1.PodRunning {
+			return microerror.Maskf(unexpectedStatusPhaseError, "current status: %s", string(phase))
+		}
+		return nil
+	}
+}
+
+func (f *framework) activeNamespaceFunc(name string) func() error {
```

Comment:
Those names are strange. Even though they are passed as arguments to other functions I think names should reflect what they do, not how they are used.

Reply:
Makes total sense, thanks for pointing it out, fixed.